assets/js/amp-editor-blocks.js

```diff
@@ -0,0 +1,596 @@
+			PanelBody = wp.components.PanelBody;
+
+		return isSelected && (
+			el( InspectorControls, { key: 'inspector' },
+				el( PanelBody, { title: component.data.ampPanelLabel },
+					component.getAmpCarouselToggle( props ),
+					component.getAmpLightboxToggle( props )
+				)
+			)
+		);
+	};
+
+	/**
+	 * Set up inspector controls for shortcode block.
+	 * Adds ampCarousel attribute in case of gallery shortcode.
+	 *
+	 * @param {Object} props Props.
+	 * @return {*} Inspector controls.
+	 */
+	component.setUpShortcodeInspectorControls = function setUpShortcodeInspectorControls( props ) {
+		var isSelected = props.isSelected,
+			el = wp.element.createElement,
+			InspectorControls = wp.blocks.InspectorControls,
+			PanelBody = wp.components.PanelBody;
+
+		if ( component.isGalleryShortcode( props.attributes ) ) {
+			return isSelected && (
+				el( InspectorControls, { key: 'inspector' },
+					el( PanelBody, { title: component.data.ampPanelLabel },
+						component.getAmpCarouselToggle( props )
+					)
+				)
+			);
+		}
+
+		return '';
+	};
+
+	/**
+	 * Filters blocks' save function.
+	 *
+	 * @param {Object} element Element.
+	 * @param {string} blockType Block type.
+	 * @param {Object} attributes Attributes.
+	 * @return {*} Output element.
+	 */
+	component.filterBlocksSave = function filterBlocksSave( element, blockType, attributes ) {
+		var text,
+			ampClassName = element.props.className || '',
+			props = element.props;
+		if ( 'core/shortcode' === blockType.name && component.isGalleryShortcode( attributes ) ) {
+			if ( attributes.ampCarousel ) {
+				// If the text contains amp-carousel, lets remove it.
+				if ( component.hasGalleryShortcodeCarouselAttribute( attributes.text || '' ) ) {
+					text = component.removeAmpCarouselFromShortcodeAtts( attributes.text );
+
+					return wp.element.createElement(
+						wp.element.RawHTML,
+						{},
+						text
+					);
+				}
+
+				// Else lets return original.
+				return element;
+			}
+
+			// If the text already contains amp-carousel, return original.
+			if ( component.hasGalleryShortcodeCarouselAttribute( attributes.text || '' ) ) {
+				return element;
+			}
+
+			// Add amp-carousel=false attribut to the shortcode.
+			text = attributes.text.replace( '[gallery', '[gallery amp-carousel=false' );
+
+			return wp.element.createElement(
+				wp.element.RawHTML,
+				{},
+				text
+			);
+		}
+
+		// In case AMP attributes, add info to classname.
+		if ( component.hasAmpLayoutSet( attributes || '' ) ) {
+			ampClassName += ' amp-layout-' + attributes.ampLayout;
+		}
+		if ( component.hasAmpNoLoadingSet( attributes || '' ) ) {
+			ampClassName += ' amp-noloading';
+		}
+		if ( component.hasAmpCarouselSet( attributes || '' ) ) {
+			ampClassName += ' amp-carousel';
+		}
+		if ( component.hasAmpLightboxSet( attributes || '' ) ) {
+			ampClassName += ' amp-lightbox';
+		}
+
+		if ( '' !== ampClassName && attributes.className !== ampClassName ) {
+			props.className = ampClassName.trim();
```

Comment:
Hi @miina,
In my browser, this didn't assign the new `props.className` value, causing the lightbox to not appear on the front-end.

But with this change (borrowing from Weston's solution [here](https://github.com/Automattic/amp-wp/blob/7439d2ab5eebd0ee071749a849fda946a2eff4be/assets/js/amp-block-validation.js#L357)), it did:

```js
props = _.extend( {}, props, { className: ampClassName.trim() } );
```

Reply:
Hey @kienstra, thanks for finding this, makes sense that the `props.className` can't be directly set this way.